metricbeat/module/vsphere/network/network_test.go

```diff
@@ -0,0 +1,93 @@
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package network
+
+import (
+	"testing"
+
+	mbtest "github.com/elastic/beats/v7/metricbeat/mb/testing"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware/govmomi/simulator"
+)
+
+func TestFetchEventContents(t *testing.T) {
+	model := simulator.VPX()
+	if err := model.Create(); err != nil {
+		t.Fatal(err)
+	}
+
+	ts := model.Service.NewServer()
+	defer ts.Close()
+
+	f := mbtest.NewReportingMetricSetV2WithContext(t, getConfig(ts))
+	events, errs := mbtest.ReportingFetchV2WithContext(f)
+	if len(errs) > 0 {
+		t.Fatalf("Expected 0 error, had %d. %v\n", len(errs), errs)
+	}
+
+	assert.NotEmpty(t, events)
```

Comment:
[Blocker]
Use `require` instead as it'll stop the test. Otherwise the test will continue and panic on https://github.com/elastic/beats/pull/40559/files#diff-a30895d791dcfe0293c7afeaaca79cb9ea151f4e6d6f1426797f6cbc659312cdR42. And Ideally also add a message to the assertion

```suggestion
	require.NotEmpty(t, events. "didn't get any event, should have gotten at least X")
```